chrome-extension/popup.js

```diff
@@ -23,79 +42,139 @@ window.onload = function() {
 
   document.getElementById('generate-note-btn').onclick = generateNote;
   document.getElementById('logout-btn').onclick = logout;
+  document.getElementById('login-btn').onclick = login;
   document.getElementById('doc-name-input').value = 'gNote ' + getDate();
 }
 
 /** Initializes gapi. */
 window.initGAPI = function initGAPI() {
   gapi.client.init({
-    apiKey: keys.API_KEY,
+    apiKey: API_KEY,
     discoveryDocs: DISCOVERY_DOCS,
-  }).then(handleLogin)
+  }).then(login)
   .catch(error => console.log('Error:', error));
 }
 
 /**
  * Requests user login (if necessary) then sets gapi access token
  * and displays user info in popup.
  */
-function handleLogin() {
-  chrome.identity.getAuthToken({interactive: true}, function(token) {
-    if (chrome.runtime.lastError) {
-      return;
+function login() {
+  const url =
+      'https://accounts.google.com/o/oauth2/auth' + 
+      '?client_id=' + CLIENT_ID + 
+      '&response_type=code' + 
+      '&access_type=offline' +
+      '&redirect_uri=' + REDIRECT_URI + 
+      '&scope=' + SCOPES;
+  chrome.identity.launchWebAuthFlow({
+    url: url,
+    interactive: true
+  }, redirectedTo => {
+    if(chrome.runtime.lastError) {
+      console.log(chrome.runtime.lastError);
```

Comment:
Typically its good to remove log statements used for debugging into the master branch.